test/cypress/integration/simplified-account/service-settings/email-notifications.cy.js

```diff
@@ -142,7 +148,72 @@ describe('email notifications settings', () => {
     })
   })
 
-  describe('edit refund email settings', () => {
+  describe('Refund email settings', () => {
+    describe('When email collection mode is MANDATORY or OPTIONAL', () => {
+      beforeEach(() => {
+        cy.task('setupStubs', [
+          gatewayAccountStubs.setRefundEmailEnabledByServiceIdAndAccountType(SERVICE_EXTERNAL_ID, ACCOUNT_TYPE, 'false')
+        ])
+      })
+
+      it('should navigate to the refund email toggle page', () => {
+        ['MANDATORY', 'OPTIONAL'].forEach(mode => {
+          setupStubs(ADMIN_ROLE, mode)
+          cy.visit(EMAIL_NOTIFICATIONS_URL)
+          cy.get('.govuk-summary-list').within(() => {
+            cy.get('.govuk-summary-list__actions a').eq(2).click()
+          })
+
+          cy.title().should('contains', 'Settings - Email notifications')
+          cy.url().should('include', '/settings/email-notifications/refund-email-toggle')
+          cy.get('.govuk-fieldset__heading').first().should('contain', 'Do you want to send refund emails?')
 
+          cy.get('.govuk-radios').within(() => {
+            cy.get('.govuk-radios__item').eq(0).should('contain', 'Yes')
+            cy.get('.govuk-radios__item').eq(1).should('contain', 'No')
+          })
+
+          cy.get('input[type="radio"][value="false"]').check()
+          cy.get('.govuk-button').contains('Save changes').click()
+          cy.get('h1').should('contain', 'Email notifications')
+          cy.title().should('eq', 'Settings - Email notifications - GOV.UK Pay')
+        })
+      })
+    })
+
+    describe('When email collection mode is OFF', () => {
+      beforeEach(() => {
+        setupStubs(ADMIN_ROLE, 'OFF')
+      })
+      it('there should be no "Change" link on the Payment confirmation emails and Refund emails rows', () => {
+        cy.visit(EMAIL_NOTIFICATIONS_URL)
+
+        cy.get('.govuk-summary-list').within(() => {
+          cy.get('.govuk-summary-list__key').eq(0).should('contain', 'Collect email addresses')
+          cy.get('.govuk-summary-list__value').eq(0).should('contain', 'No')
+          cy.get('.govuk-summary-list__actions a').eq(0).should('contain', 'Change')
+          cy.get('.govuk-summary-list__actions a.govuk-link').eq(0).should('have.attr', 'href',
+            `/simplified/service/${SERVICE_EXTERNAL_ID}/account/test/settings/email-notifications/email-collection-mode`)
+
+          // Verify there's only one change link. It's currently not possible to verify there is no 'Change' link on the 1st and 2nd row
```

Comment:
can't this be done by having a specific id on those links that cypress can look for?

Reply:
Unfortunately not as this is a list generated by the [govukSummaryList design system](https://design-system.service.gov.uk/components/summary-list/). There is no option to generate unique ids per row item. As we've already verified there's a link with `/simplified/service/${SERVICE_EXTERNAL_ID}/account/test/settings/email-notifications/email-collection-mode`, and verified there's only one "Change" link I'm of the opinion this is good for now.